apps/docs/src/content/reference/extras/instanced-sprite.mdx

```diff
@@ -0,0 +1,454 @@
+
+
+## Spritesheets
+
+### SpritesheetMetadata
+
+Object used in `buildSpritesheet` function has to be compliant with the `SpritesheetMetadata` type format. This type is structured to
+accommodate the metadata for one or multiple sprite files within a single spritesheet.
+
+```ts
+type SpritesheetMetadata = {
+  url: string
+  type: 'rowColumn' | 'frameSize'
+  width: number
+  height: number
+  animations: {
+    name: string
+    frameRange: [number, number]
+  }[]
+}[]
+
+```
+
+#### Understanding SpritesheetMetadata
+
+A `SpritesheetMetadata` is an array, with each entry representing metadata fields for one sprite:
+
+	- `url`: Specifies the path or URL to the sprite image file.
+	- `type`: Determines the method of defining the spritesheet dimensions. Type `"rowColumn"` specifies the layout in terms of rows
+	and columns within the image, and type `"frameSize"` instead defines the size of each frame, allowing the utility to calculate the layout.
+	- `width` and `height`: Depending on type, these refer to the number of `columns` and `rows` (`"rowColumn"`) or the dimensions of a single frame (`"frameSize"`).
+	- `animations`: An array detailing the animations, where each animation has a name and a `frameRange`. The `frameRange` is a tuple
+	marking the start and end frames of the animation.
+
+#### Typesafety
+
+For improved developer experience when working with TypeScript, it is strongly recommended to use `as const` assertion in
+combination with `satisfies SpritesheetMetadata`. This approach not only ensures compliance with the `SpritesheetMetadata`
+type but also enables autocompletion for animation names within utility functions, which is highly recommended:
+
+
+### `buildSpritesheet()`
+
+`buildSpritesheet()` is a utility function for building a final texture and spritesheet object from a provided `SpritesheetMetadata` object or external source.
+Each `buildSpritesheet` method return an Promise that and has to be awaited. Promise returned by each method contains an object with a `spritesheet`
+ready for use in `<InstancedSprite>`.
+
+#### buildSpritesheet().from(meta: SpritesheetMetadata)
+Other than `spritesheet` promise, it also returns a `useInstancedSprite` hook. This hook can be enhanced with
+extra typescript support for autocompletion of animation names as such:
+
+
+```ts
+ const meta = [
+    {
+      url: '/textures/sprites/cacodaemon.png',
+      type: 'rowColumn',
+      width: 8,
+      height: 4,
+      animations: [
+        { name: 'fly', frameRange: [0, 5] },
+        { name: 'attack', frameRange: [8, 13] },
+        { name: 'idle', frameRange: [16, 19] },
+        { name: 'death', frameRange: [24, 31] }
+      ]
+    }
+  ] as const satisfies SpritesheetMetadata
+
+  const result = buildSpritesheet.from<typeof meta>(meta)
+```
+
+![Tree sprite atlas](/images/docs/extras/instanced-sprite/useInstancedSpriteAutocomplete.png)
+
+#### buildSpritesheet().fromAseprite(asepriteDataUrl: string, spriteImageUrl: string)
```

Comment:
Same here. Don't use code snippets as titles, unless its a simple function / class name.

Do keep this snippet as a usage example, but put it in an inline or self-standing code block in the section content like a sane person.